src/zenml/utils/artifact_utils.py

```diff
@@ -427,6 +433,30 @@ def upload_artifact(
     return response.id
 
 
+def _get_new_artifact_version(artifact_name: str) -> str:
+    """Get the next auto-incremented version for an artifact name.
+
+    Args:
+        artifact_name: The name of the artifact.
+
+    Returns:
+        The next auto-incremented version.
+    """
+    models = depaginate(
+        partial(
+            Client().list_artifacts,
+            name=artifact_name,
+            sort_by="desc:version",
+        )
+    )
```

Comment:
This can be heavy, maybe adding some extra helper store function to retrieve the maximum numerical name would be a better way. This should be achievable with SQLModels interface.
If you go with Number in DB - ignore this.

Reply:
yeah I think I will go with number in DB 👍